cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -0,0 +1,179 @@
+				],
+			);
+
+			assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::ForeignAssets::force_create(
+				RuntimeOrigin::root(),
+				asset_location.clone(),
+				asset_hub_sovereign.clone().into(),
+				false,
+				1,
+			));
+
+			assert!(<AssetHubWestend as AssetHubWestendPallet>::ForeignAssets::asset_exists(
+				asset_location.clone().try_into().unwrap(),
+			));
+		}
+	});
+
+	let native_tokens: Vec<snowbridge_router_primitives::inbound::v2::Asset> = token_ids
+		.iter()
+		.map(|token_id| NativeTokenERC20 { token_id: *token_id, value: 3_000_000_000 })
+		.collect();
+
+	BridgeHubWestend::execute_with(|| {
+		type RuntimeEvent = <BridgeHubWestend as Chain>::RuntimeEvent;
+
+		let claimer = AccountId32 { network: None, id: receiver.clone().into() };
+		let claimer_bytes = claimer.encode();
+		let origin = H160::random();
+		let relayer_location =
+			Location::new(0, AccountId32 { network: None, id: relayer.clone().into() });
+
+		let message_xcm_instructions =
+			vec![DepositAsset { assets: Wild(AllCounted(8)), beneficiary: receiver.into() }];
+		let message_xcm: Xcm<()> = message_xcm_instructions.into();
+		let versioned_message_xcm = VersionedXcm::V5(message_xcm);
+
+		let message = Message {
+			origin,
+			assets: native_tokens,
+			xcm: versioned_message_xcm.encode(),
+			claimer: Some(claimer_bytes),
+		};
+		let xcm = EthereumInboundQueueV2::do_convert(message, relayer_location).unwrap();
+		let _ = EthereumInboundQueueV2::send_xcm(xcm, AssetHubWestend::para_id().into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubWestend,
+			vec![RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},]
+		);
+	});
+
+	let execution_fee = WeightCalculator::weight_to_fee(&Weight::from_parts(1410450000, 33826));
+	let buffered_fee = execution_fee * 2;
+	println!("buffered execution fee for prologue for 8 assets: {}", buffered_fee);
+}
+
+#[test]
+fn register_token_xcm() {
+	BridgeHubWestend::execute_with(|| {
+		println!("register token mainnet: {:x?}", get_xcm_hex(1u64));
+		println!("===============================",);
+		println!("register token sepolia: {:x?}", get_xcm_hex(11155111u64));
+	});
+}
+
+fn get_xcm_hex(chain_id: u64) -> String {
+	let owner = EthereumLocationsConverterFor::<[u8; 32]>::from_chain_id(&chain_id);
+	let weth_token_id: H160 = hex!("be68fc2d8249eb60bfcf0e71d5a0d2f2e292c4ed").into(); // TODO insert token id
+	let token: H160 = hex!("c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2").into(); // token id placeholder
+	let weth_amount = 300_000_000_000_000u128;
+
+	let ethereum_network_v5: NetworkId = EthereumNetwork::get().into();
+	let dot_asset = Location::new(1, Here);
+	let dot_fee: xcm::prelude::Asset = (dot_asset, CreateAssetDeposit::get()).into();
+
+	println!("register token id: {:x?}", token);
+	println!("weth token id: {:x?}", weth_token_id);
+	println!("weth_amount: {:x?}", hex::encode(weth_amount.encode()));
+	println!("dot asset: {:x?}", hex::encode(dot_fee.encode()));
```

Comment:
also need the DOT amount

Reply:
Do you want to specify the DOT amount in the contract as well? That should stay constant given the runtime config does not change.

Reply:
yeah that would be better